common/primitives/timestamp/time.go

```diff
@@ -207,20 +207,32 @@ func RoundUp(d time.Duration) time.Duration {
 	return res + time.Second
 }
 
-func UnixOrZeroTime(nanos int64) time.Time {
+func UnixOrEmptyTime(nanos int64) time.Time {
 	if nanos <= 0 {
 		return time.Time{}
 	}
 
+	return UnixOrZeroTime(nanos)
+}
+
+func UnixOrEmptyTimePtr(nanos int64) *time.Time {
+	return TimePtr(UnixOrEmptyTime(nanos))
+}
+
+func UnixOrZeroTime(nanos int64) time.Time {
```

Comment:
`UnixOrZeroTime` to be renamed to `UnixNano` in later PR - `UnixOrEmptyTime` will now return empty time object.